linera-views/src/common.rs

```diff
@@ -369,6 +360,48 @@ pub trait WritableKeyValueStore<E> {
     async fn clear_journal(&self, base_key: &[u8]) -> Result<(), E>;
 }
 
+/// Low-level, asynchronous write and read key-value operations. Useful for storage APIs not based on views.
+#[async_trait]
+pub trait AdminKeyValueStore<E>: Sized {
+    /// The configuration needed to interact with a new store.
+    type Config: Send + Sync;
+
+    /// Connects to an existing namespace using the given configuration.
+    async fn connect(config: &Self::Config, namespace: &str) -> Result<Self, E>;
+
+    /// Obtains the list of existing namespaces.
+    async fn list_all(config: &Self::Config) -> Result<Vec<String>, E>;
+
+    /// Delete all the existing namespaces.
+    async fn delete_all(config: &Self::Config) -> Result<(), E>;
```

Comment:
Should we provide a default implementation?

Reply:
We could of course, though the implementation for ScyllaDB uses a specific trick of that database.

Reply:
That's why default implementations can be overridden!

Reply:
How interesting!
I did not know that, though that features feels very C++y and not in a good way.

I added a default implementation and used it for memory only.